pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1200,13 +1210,16 @@ func fetchRegistrationEntry(ctx context.Context, db *sqlDB, req *datastore.Fetch
 	}, nil
 }
 
-func buildFetchRegistrationEntryQuery(dbType string, req *datastore.FetchRegistrationEntryRequest) (string, []interface{}, error) {
+func buildFetchRegistrationEntryQuery(dbType string, supportsCTE bool, req *datastore.FetchRegistrationEntryRequest) (string, []interface{}, error) {
 	switch dbType {
 	case SQLite:
 		return buildFetchRegistrationEntryQuerySQLite3(req)
 	case PostgreSQL:
 		return buildFetchRegistrationEntryQueryPostgreSQL(req)
```

Comment:
Is this because postgres and sqlite "always" supports CTE?

We are having two spots (per DB type) in which the code "knows" that CTE is available - once in the per-type `connect` function, and once here. It's not super clear from just reading the code that this was in fact intentional and not an oversight...

Comments in this area, or moving build* functions to unconditionally accept supportsCTE bool (and subsequent comments in those handling functions), or something along those lines I think will make it clear to the reader that this was not some patch that MySQL received but others missed out on.

Reply:
This whole "switch on db type" business is gross. I started down a refactor to clean up some of these interactions so there would be no branching in the RPC handlers around type but decided to back it out:

From my earlier comment:
> I had to resist doing a larger refactor to the code to make it less branchy since we'll probably be cherry-picking this into a 0.9.3 release branch.